tests/components/sensor/test_arlo.py

```diff
@@ -0,0 +1,185 @@
+"""The tests for the Netgear Arlo sensors."""
+import asyncio
+from collections import namedtuple
+import unittest
+from unittest.mock import patch, MagicMock
+from homeassistant.const import (
+    DEVICE_CLASS_TEMPERATURE, DEVICE_CLASS_HUMIDITY, ATTR_ATTRIBUTION)
+from homeassistant.components.sensor import arlo
+from homeassistant.components.arlo import DATA_ARLO
+from homeassistant.helpers import dispatcher
+from tests.common import get_test_home_assistant
+
+
+class TestArloSensor(unittest.TestCase):
+    """Test Netgear Arlo sensors."""
+
+    def _setup_arlo(self):
+        self.hass.data = {
+            DATA_ARLO: 'test'
+        }
+
+    @staticmethod
+    def _get_mock_sensor(name='Last', sensor_type='last_capture', data=None):
+        if data is None:
+            data = {}
+        return arlo.ArloSensor(name, data, sensor_type)
+
+    @staticmethod
+    def _get_named_tuple(input_dict):
+        return namedtuple('Struct', input_dict.keys())(*input_dict.values())
+
+    def setUp(self):  # pylint: disable=invalid-name
+        """Setup shared dependencies for each test."""
+        self.hass = get_test_home_assistant()
+
+    def tearDown(self):  # pylint: disable=invalid-name
+        """Tear down shared dependencies for each test."""
+        self.hass.stop()
+
+    def test_setup_with_no_data(self):
+        """Test setup_platform with no data."""
+        result = arlo.setup_platform(self.hass, None, None)
+        self.assertEqual(result, False)
+
+    sensors = None
+
+    def _add_devices(self, sensors, boolean):
+        if boolean:
```

Comment:
Why would we only add the sensors when passing `True` as the second argument? That's not how the actual code works.

Reply:
Not actually sure what the second param does -- just noticed that it's hardcoded `True` in the sensor code. But this does make the test rather brittle and potentially (probably) incorrect. I've removed the condition for now and will take another pass after some discovery.

Reply:
Passing `True` to `add_devices` calls `entity.update` during entity addition.